src/components/DataTable/TableSelectRow.js

```diff
@@ -16,19 +17,26 @@ const TableSelectRow = ({
   name,
   onSelect,
   disabled,
-}) => (
-  <td>
-    <InlineCheckbox
-      id={id}
-      name={name}
-      onClick={onSelect}
-      checked={checked}
-      ariaLabel={ariaLabel}
-      disabled={disabled}
-    />
-  </td>
-);
-
+  radio,
+}) => {
+  const selectionInputProps = {
+    id,
+    name,
+    onClick: onSelect,
+    checked,
+    ariaLabel,
+    disabled,
+  };
+  return (
+    <td>
+      {radio ? (
```

Comment:
How about `const CheckmarkComponent = radio ? RadioButton : InlineCheckbox` so we don't need to change the structure here...?

Reply:
Thanks @emyarod for the update - Do you think we still need [`selectionInputProps` variable](https://github.com/IBM/carbon-components-react/pull/1824/files#diff-bf0d07e8541d1f68866a05a894ee7af4R22)...?

Reply:
@asudoh I'm not entirely sure if we still need it, but we might just leave it so that we don't spread any additional props into the `<InlineInputComponent>` and for assigning `onClick` to `onSelect`